ERCS/erc-7654.md

```diff
@@ -0,0 +1,182 @@
+
+    /**
+     * Request the contract to retrieve records.
+     * @param _methodName is the method name.
+     * @param _methodReq is the method type.
+     * @return The response to the get request.
+     */
+    function get(string memory _methodName,bytes memory _methodReq)external view returns(bytes memory);
+
+    /**
+     * Request the contract to create a new record.
+     * @param _methodName is the method name.
+     * @param _methodReq is the method type.
+     * @return The response to the post request.
+     */
+    function post(string memory _methodName,bytes memory _methodReq)external returns(bytes memory);
+
+    /**
+     * Request the contract to update a record.
+     * @param _methodName is the method name.
+     * @param _methodReq is the method type.
+     * @return The response to the put request.
+     */
+    function put(string memory _methodName,bytes memory _methodReq)external returns(bytes memory);
+
+    /**
+     * Supported request method types.
+     * @return Method types.
+     */
+    function options()external returns(MethodTypes[] memory);
+}
+
+```
+
+### Library
+
+The library [`Types.sol`](../assets/eip-7654/Types.sol) contains an enumeration of Solidity types used in the above interfaces.
+
+## Rationale
+
+### How to customize response status code?
+
+The return value type of each request method is bytes, so you can customize the response status code and other response information according to your needs.
+
+### How to disable a request method?
+
+You can add a mapping(string=>bool) to save the request method name and available state. After the application is upgraded, manually set the available state of the disabled request method to false. And check the state every time a request is processed.
+
+### Why is there no delete method type?
+
+The data in smart contracts is public, and deleting data is an inefficient operation. In order to facilitate data management and retrieval, you can add a mapping to save the valid state of the data, and add a put request method to set valid and invalid data. And only return valid data in get method.
+
+### How to dynamically add new request methods?
+
+Before deploying the contract, you can add a mapping (address=>bool) to record the external contract permissions. When the application adds new functions, give the v2 contract write permissions and add new request methods to the v2 contract.
+
+### How to encode request parameters and decode response?
+
+Use ```getMethodReqAndRes``` to get the data type of the method request parameters and the data type of the response value. Then use some js library to encode and decode it.   
+For example:   
+
+```javascript
+var reqDataEncode = web3.eth.abi.encodeParameters(
+  ["string", "uint256"],
+  ["alice", "1"]
+);
+
+var resDataDecode = web3.eth.abi.decodeParameters(
+  ["string", "uint256"],
+  reqDataEncode
+);
+```
```

Comment:
Most of your Rationale reads like implementation hints, and less like the reasoning behind your design decisions. The Rationale section is supposed to explain technical choices you made within the document.